pkg/patch/internal/exec.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"fmt"
+
+	exec "k8s.io/utils/exec"
+)
+
+//counterfeiter:generate . Cmd
+type Cmd = exec.Cmd
+
+type CommandCreator func(path string, args ...string) Cmd
+
+func (c CommandCreator) create(path string, args ...string) Cmd {
+	if c == nil {
```

Comment:
can `create()` be called multiple times on the same object?
it might be better to guard for this on the side of the callers?

Reply:
- technically, `create` can be called multiple times. I don't think here anything calls it multiple times, but I don't see a reason why it could or should not be done.
- the `nil` check is there, because I appreciate the `nil` receiver.  If users have a thing which is of the interface type `CommandCreator`, but have not set it to anything (ie. it is `nil`), things still work without forcing users to do `nil` checks. To rephrase that: users can use a thing of type `CommandCreator` anywhere, don't need to initialise it but can just call `create` on it and it will work; they are however also free to overwrite the thing and change the implementation. This allows to just use the zero type of structs which hold a thing of type `CommandCreator`.
- I put most of the stuff into the `internal` package to keep the external facing API to a minimum. I could see certain things to be useful for the whole module or even beyond. If we'd decide to move things around we probably also need to export certain things, e.g. the `create` method here specifically.

Reply:
i'd prefer this to panic on user calls with nil objects, but this pattern is also present in k/k.
i complained to the people there too. :old-man-yells-at-cloud: